From c00d76ed3e399c4ea6f3050c0d53b3fae730c3b0 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Sat, 13 Oct 2012 16:34:44 +0300
Subject: [PATCH 02/32] omap: Export
 omap_hwmod_lookup/omap_device_build/omap_device_build_ss

These functions can be used just fine by modules, there's no need not
to have them exported.
---
 arch/arm/mach-omap2/omap_hwmod.c |    2 ++
 arch/arm/plat-omap/omap_device.c |    6 ++++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod.c b/arch/arm/mach-omap2/omap_hwmod.c
index 87cc6d0..f3c0947 100644
--- a/arch/arm/mach-omap2/omap_hwmod.c
+++ b/arch/arm/mach-omap2/omap_hwmod.c
@@ -138,6 +138,7 @@
 #include <linux/spinlock.h>
 #include <linux/slab.h>
 #include <linux/bootmem.h>
+#include <linux/export.h>
 
 #include <plat/clock.h>
 #include <plat/omap_hwmod.h>
@@ -3076,6 +3077,7 @@ struct omap_hwmod *omap_hwmod_lookup(const char *name)
 
 	return oh;
 }
+EXPORT_SYMBOL(omap_hwmod_lookup);
 
 /**
  * omap_hwmod_for_each - call function for each registered omap_hwmod
diff --git a/arch/arm/plat-omap/omap_device.c b/arch/arm/plat-omap/omap_device.c
index 7a7d1f2..a15b715 100644
--- a/arch/arm/plat-omap/omap_device.c
+++ b/arch/arm/plat-omap/omap_device.c
@@ -663,7 +663,7 @@ void omap_device_delete(struct omap_device *od)
  * information.  Returns ERR_PTR(-EINVAL) if @oh is NULL; otherwise,
  * passes along the return value of omap_device_build_ss().
  */
-struct platform_device __init *omap_device_build(const char *pdev_name, int pdev_id,
+struct platform_device *omap_device_build(const char *pdev_name, int pdev_id,
 				      struct omap_hwmod *oh, void *pdata,
 				      int pdata_len,
 				      struct omap_device_pm_latency *pm_lats,
@@ -678,6 +678,7 @@ struct platform_device __init *omap_device_build(const char *pdev_name, int pdev
 				    pdata_len, pm_lats, pm_lats_cnt,
 				    is_early_device);
 }
+EXPORT_SYMBOL(omap_device_build);
 
 /**
  * omap_device_build_ss - build and register an omap_device with multiple hwmods
@@ -696,7 +697,7 @@ struct platform_device __init *omap_device_build(const char *pdev_name, int pdev
  * platform_device record.  Returns an ERR_PTR() on error, or passes
  * along the return value of omap_device_register().
  */
-struct platform_device __init *omap_device_build_ss(const char *pdev_name, int pdev_id,
+struct platform_device *omap_device_build_ss(const char *pdev_name, int pdev_id,
 					 struct omap_hwmod **ohs, int oh_cnt,
 					 void *pdata, int pdata_len,
 					 struct omap_device_pm_latency *pm_lats,
@@ -751,6 +752,7 @@ odbs_exit:
 
 	return ERR_PTR(ret);
 }
+EXPORT_SYMBOL(omap_device_build_ss);
 
 /**
  * omap_early_device_register - register an omap_device as an early platform
-- 
1.7.7.6

